src/vs/workbench/contrib/notebook/browser/view/cellParts/cellOutput.ts

```diff
@@ -689,6 +683,12 @@ export class CellOutputContainer extends CellContentPart {
 		// shrink immediately as the final output height will be zero.
 		// if it's rerun, then the output clearing might be temporary, so we don't shrink immediately
 		this._validateFinalOutputHeight(context === CellOutputUpdateContext.Other && this.viewCell.outputsViewModels.length === 0);
+		// after splice, it doesn't exceed
+		deletedEntries.forEach(entry => {
+			entry.element.detach();
+			entry.element.dispose();
```

Comment:
I have tried to move the code that detaches the disposes the previous output items to the very bottom,
so that it happens very late, preferably after rendering the new items,

Reply:
Previously the old itemss were not getting detached/disposed, and thats why the rendering was very smooth,